src/cpp/rtps/writer/ReaderProxy.cpp

```diff
@@ -501,12 +501,12 @@ bool ReaderProxy::perform_nack_supression()
     return convert_status_on_all_changes(UNDERWAY, UNACKNOWLEDGED);
```

Comment:
This should be done wherever `convert_status_on_all_changes` is used

```suggestion
    return 0 != convert_status_on_all_changes(UNDERWAY, UNACKNOWLEDGED);
```